workflow.md

```diff
@@ -40,50 +52,61 @@
+  I am not sure this is correct, let's discuss. I remember there was a
+  list of cases, but I can't find it of course.
 
 ## Pull requests
 
-When you finish your work in your branch, then you should create a pull request
-to your project. Create a PR to `master` branch (or any other branch if you were
-explicitly asked). Please assign a reviewer (if you don't know a person then
-project manager would do it himself). Do not merge your own PRs!
+When you finish your work in your branch, you should create a pull
+request to your project. Create a PR to `master` branch or another
+appropriate branch. Please assign a reviewer (if you don't know a
+person then project manager would do it himself). You can merge your
+own PR only if it's approved and all mandatory CI checks pass.
 
 Please describe in PR what have you done, if it is tested or not, any additional
 information.
 
+You can also create a PR before your work is finished, but in this
+case please add a `wip` label to it.
+
 ## Merging Pull requests
 
-After review by another team member, the teammate merging the PR should merge
-`master` into the pull request's branch and then the PR branch into `master`,
-with the `--no-ff` option enabled.
+After review by another team member, the teammate merging the PR
+should merge `master` into the pull request's branch to resolve
+conflicts and then the PR branch into `master`, with the `--no-ff`
+option enabled. Usually it's enough to just press `Merge` button on
+github, as long as all checks pass and there are no conflicts.
 
-Generally, the pull request's branch should be merged into master , but if is
-clear that rebasing this branch onto master is better (for instance, it is a
-trivial change which doesn't deserve branching) then a rebase can be done
-instead.
+Generally, the pull request's branch should be merged into master,
+but if it is clear that rebasing this branch onto master is better
+(for instance, it is a trivial change which doesn't deserve branching)
+then a rebase can be done instead. Use your judgement.
 
 ## Versions
 
 Versions are denoted with 3 numbers: `Maj.Min.Fix`.
 
 * `Maj` is major version which changes rarely, when something very
-  significant is changed, it can possibly break API. It may be useful
-  for marketing.
-  * Example: hardfork in `cardano-sl`.
-* `Min` is minor version, changes when new features are added,
-  but they don't break API
-  and are not so important from marketing point of view.
-  * Example: add new endpoints to server.
+  significant is changed, it can possibly break all kinds of API it
+  provides. It may be useful for marketing.
+  * Example: input endorsers are added to `cardano-sl`.
+* `Min` is minor version, changes when new features are added, can
+  break API but tries not to break too much.
+  * Example: add new endpoints to a server.
 * `Fix` is fix version, can be changed only because of bugfix
   or when something minor is added, insignificant from end-user perspective.
```

Comment:
I think it is hard to say that `feature A` is a minor change (`Fix`) and `feature B` is not (`Min`). That might be subjective. What about to mark any feature (minor, small, large) w/o breaking changes as `Min`? Just to simplify any personal decision.